### DISCUSSION TITLE
Error when developing a shiny app with quarto

### DISCUSSION THREAD

**Discussion by ManuelSpinola** [quarto-dev/quarto-cli#5773] [Q&A] [answered]
### Description

I am trying to develop a shiny app for Species locations.  Here is a reproducible example with the New Zeland data from the spData package.

After running the quarto document I got the following message:

Error: nz does not contain data

````
---
title: "New Zeland"
format: 
  html:
    page-layout: custom
server: shiny
---

```{r}
#| context: setup
#| mesaage: false
#| warning: false
library(shiny)
library(rmarkdown)
library(tidyverse)
library(rio)
library(sf)
library(mapview)
library(leaflet)
library(spData)
```

```{r}
#| context: setup
nz |>
  st_transform(4326)
```


```{r}
#| panel: sidebar
selectInput("nz", label = "Region", choices = unique(nz$Island))
```

```{r}
#| panel: fill
leafletOutput('plot1', height = "80vh")
```

```{r}
#| context: server

nzInput <- reactive({
  nz[nz$Island == input$Island, ]
})

output$plot1 <- renderLeaflet({
  nz <- nzInput()
  m <- mapview(nz)
  m@map
})
```
````

**Comment by mcanouil** [comment, marked as answer]
Your issue is not related to Quarto.
You should consider first trying your app as an app before integrating it into Quarto.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/b9afcebf-2cdf-49b0-9c74-8e28b8d6fd11">

This being said, `input$Island` is not valid, because from your code, the only element in `input` is "nz".
By the way, naming, the id of the input "nz", the dataset "nz", and the reactive value of the input "nz", does not seem to be the best idea for debugging, _i.e._, you can't really tell from an error message "nz" refers to what.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/771d7a72-368c-4ac9-a739-75add9f82844">

**Reply by mcanouil** [reply]
Closing this as not relevant to Quarto.

**Comment by cderv** [comment]
@ManuelSpinola You can ask for help on Shiny at https://community.rstudio.com/ for example.

**Comment by ManuelSpinola** [comment]
Thank you very much.

El jue, 1 jun 2023 a las 4:00, Christophe Dervieux (<
***@***.***>) escribió:

> @ManuelSpinola <https://github.com/ManuelSpinola> You can ask for help on
> Shiny at https://community.rstudio.com/ for example.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/quarto-dev/quarto-cli/discussions/5773#discussioncomment-6059126>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFI3FB5P46BAP3TOXCFGMZ3XJBR5DANCNFSM6AAAAAAYWJRVXI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Manuel Spínola, Ph.D.*
Instituto Internacional en Conservación y Manejo de Vida Silvestre
Universidad Nacional
Apartado 1350-3000
Heredia
COSTA RICA
***@***.*** ***@***.***>
***@***.***
Teléfono: (506) 8706 - 4662
Sitio web institucional: ICOMVIS
<http://www.icomvis.una.ac.cr/index.php/manuel>
Sitio web personal: Sitio personal <https://mspinola-sitioweb.netlify.app>
Blog sobre Ciencia de Datos: Blog de Ciencia de Datos
<https://mspinola-ciencia-de-datos.netlify.app>
